parallel-install/pkg/preinstaller/preinstaller.go

```diff
@@ -0,0 +1,127 @@
+package preinstaller
+
+import (
+	"fmt"
+	"github.com/avast/retry-go"
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/config"
+	"io/ioutil"
+	"k8s.io/client-go/dynamic"
+	"os"
+)
+
+// PreInstaller prepares k8s cluster for Kyma installation.
+type PreInstaller struct {
+	applier       ResourceApplier
+	cfg           config.Config
+	dynamicClient dynamic.Interface
+	retryOptions  []retry.Option
+}
+
+// NewPreInstaller creates a new instance of PreInstaller.
+func NewPreInstaller(applier ResourceApplier, cfg config.Config, dynamicClient dynamic.Interface, retryOptions []retry.Option) *PreInstaller {
+	return &PreInstaller{
+		applier:       applier,
+		cfg:           cfg,
+		dynamicClient: dynamicClient,
+		retryOptions:  retryOptions,
+	}
+}
+
+// InstallCRDs on a k8s cluster.
+// Returns Output containing results of installation.
+func (i *PreInstaller) InstallCRDs() (Output, error) {
```

Comment:
Do we consider "UninstallCRDs" operation? I think uninstall should be supported by the library.

Reply:
So far only logic of resources installation (such as crds or namespaces) has been considered in scope of this task, as this is what was previously done by installer job. I haven't seen anything connected with deinstallation. Tbh your suggestion is a matter of broader Kyma "uninstallation" topic. Even though it was not mentioned in the acceptance criteria I will verify it with PO.

Reply:
The parallel installation API does offer kyma deletion: https://github.com/kyma-incubator/hydroform/blob/6d2f2ab07d912e5a880c858a502ef17e8fddc21d/parallel-install/pkg/deployment/deletion.go#L26

Therefore I guess removing the CRDs needs to be supported.

Reply:
@clebs I have created a follow-up issue for this case - https://github.com/kyma-incubator/hydroform/issues/214

Reply:
I am not sure this is fine @mjakobczyk. This PR just broke the current contract of the parallel installation API.
This time it is more or less "OK" t do this because this library is still experimental.

Nevertheless, this will be running everywhere very soon and making breaking changes like this will not be an option.
Please always consider what the current API offers and how changes affect consumers.